### PR TITLE
Version 5.0: drop Python 2, add CLI tests

### PR DESCRIPTION
**Pull request by rspeer**

### CODE REVIEW

File: ftfy/__init__.py [hunk 1]
```diff
@@ -340,10 +340,8 @@ def guess_bytes(bstring):
         return bstring.decode('utf-16'), 'utf-16'
 
     byteset = set(bytes(bstring))
```
Comment (alin-luminoso) [inline ftfy/__init__.py]: Is the inner `bytes()` here still necessary?

File: ftfy/fixes.py [hunk 2]
```diff
@@ -519,11 +516,13 @@ def remove_bom(text):
...
 
-    >>> print(remove_bom("\ufeffWhere do you want to go today?"))
+    >>> print(remove_bom(chr(0xfeff) + "Where do you want to go today?"))
     Where do you want to go today?
     """
-    return text.lstrip(unichr(0xfeff))
+    return text.lstrip(chr(0xfeff))
+
 
+UNSAFE_PRIVATE_USE_RE = re.compile('[\U00100000-\U0010ffff]')
```
Comment (rspeer) [inline ftfy/fixes.py]: I just noticed that this line snuck back in from long ago.

File: ftfy/__init__.py (hunk 1 above)
Comment (rspeer) [inline ftfy/__init__.py]: Oh right, it isn't.

File: ftfy/__init__.py [hunk 3]
```diff
@@ -195,6 +193,7 @@ def fix_text(text,
 
 def fix_file(input_file,
              encoding=None,
+             *,
```
Comment (alin-luminoso) [inline ftfy/__init__.py]: Is there a reason this `*` is after `encoding`?

File: ftfy/badness.py [hunk 4]
```diff
@@ -58,6 +56,14 @@ def _make_weirdness_regex():
...
     groups.append('[AaC][Ll]')
 
+    # Match IPA letters next to capital letters.
+    #
+    # IPA uses lowercase letters only. Some accented capital letters next to
+    # punctuation can accidentally decode as IPA letters, and an IPA letter
+    # appearing next to a capital letter is a good sign that this happened.
+    groups.append('[LA]i')
+    groups.append('i[LA]')
+
```
Comment (alin-luminoso) [inline ftfy/badness.py]: I'm so confused... didn't I review and merge this change already?

File: ftfy/chardata.py [hunk 5]
```diff
@@ -70,11 +67,9 @@ def _build_utf8_punct_regex():
     #
     # Unrelatedly, the expression that generates these bytes will be so much
     # prettier when we deprecate Python 2.
```
Comment (alin-luminoso) [inline ftfy/chardata.py]: Obsolete comment.

File: ftfy/chardata.py [hunk 6]
```diff
@@ -70,11 +67,9 @@ def _build_utf8_punct_regex():
...
-    continuation_char_list = ''.join(
-        unichr(i) for i in range(0x80, 0xc0)
-    ).encode('latin-1')
+    continuation_bytes = bytes(range(0x80, 0xc0))
```
Comment (alin-luminoso) [inline ftfy/chardata.py]: Inline this, perhaps?  (I'm trying to figure out exactly what the obsolete comment above was talking about, and maybe that's the point.)

File: ftfy/badness.py (hunk 4 above)
Comment (rspeer) [inline ftfy/badness.py]: Rebasing I guess ¯\_(ツ)_/¯

File: ftfy/__init__.py (hunk 3 above)
Comment (rspeer) [inline ftfy/__init__.py]: To make the set of keyword-only arguments the same as `fix_text`. And it's reasonable for someone to have given the encoding as a positional argument.

Comment (alin-luminoso) [inline ftfy/__init__.py]: Ah, okay.

**Pull request merged**